hydrolib/core/dflowfm/ini/util.py

```diff
@@ -622,3 +623,48 @@ def rename_keys_for_backwards_compatibility(
                 break
 
     return values
+
+
+class UnknownKeywordErrorManager:
+    """
+    Error manager for unknown keys.
+    Detects unknown keys and manages the Error to the user.
+    """
+
+    def raise_error_for_unknown_keywords(
+        self,
+        data: Dict[str, Any],
+        section_header: str,
+        fields: Dict[str, Any],
+        excluded_fields: Set,
+    ):
+        """
+        Notify the user of unknown keywords.
+
+        Args:
+            data (Dict[str, Any])   : Input data containing all set properties which are checked on unknown keywords.
+            section_header (str)    : Header of the section in which unknown keys might be detected.
+            fields (Dict[str, Any]) : Known fields of the section.
+            excluded_fields (Set)   : Fields which should be excluded from the check for unknown keywords.
+        """
+        unknown_keywords = self._get_all_unknown_keywords(data, fields, excluded_fields)
+
+        if len(unknown_keywords) == 0:
+            return
+        
+        raise ValueError(f"Unknown keywords are detected in section: '{section_header}', '{unknown_keywords}'")
+
+    def _get_all_unknown_keywords(
+        self, data: Dict[str, Any], fields: Dict[str, Any], excluded_fields: Set
+    ) -> List[str]:
+        list_of_unknown_keywords = []
+        for name, _ in data.items():
```

Comment:
You can use: `for name in data`. There is no need to get the keys and values!

Reply:
will updated